sdk/eventhub/Azure.Messaging.EventHubs/src/Metadata/PartitionConsumerMetrics.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.EventHubs.Metadata
+{
+    /// <summary>
+    ///   A set of information about the state of a partition, as observed by the consumer.
+    /// </summary>
+    ///
+    public class PartitionConsumerMetrics
+    {
+        /// <summary>
+        ///   The name of the Event Hub where the partitions reside, specific to the
+        ///   Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubName { get; }
+
+        /// <summary>
+        ///   The identifier of the partition, unique to the Event Hub which contains it.
+        /// </summary>
+        ///
+        public string Id { get; }
+
+        /// <summary>
+        ///   The sequence number of the last observed event to be enqueued in the partition.
+        /// </summary>
+        ///
+        public long? LastEnqueuedSequenceNumber { get; private set; }
+
+        /// <summary>
+        ///   The offset of the last observed event to be enqueued in the partition.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   The offset is the relative position for event in the context of the stream.  The offset
+        ///   should not be considered a stable value, as the same offset may refer to a different event
+        ///   as events reach the age limit for retention and are no longer visible within the stream.
+        /// </remarks>
```

Comment:
Isn't the sequence number subject to the same problem?

Reply:
Nope; sequence number is guaranteed unique within the context of the event stream.   It is not globally unique nor unique to all sequence numbers to ever appear, but you won't see it change when events age out nor when new events are enqueued.

That said, I probably should remove this, as the service team insists that their offset is stable.   I don't understand how, honestly, but they know better than I.

Reply:
Maybe not exactly the same problem, but can't we have the same sequence number for two different events?

Reply:
Got it. So an offset can change in an event that already exists. That's what I was confused about.

Reply:
In most systems, yes.   The offset can change as events age-out.    Let's say we have 5 events.   The first is offset 0, the second is offset 1 ... and so on.   If the first one hits 7 days old and ages out so it is no longer part of the stream.   Now Event 2 is offset 0, and everything moves down by one.  That's how I've always understood it.   Apparently, this isn't true for Event Hubs through some magic that I don't understand.

Same set of events, but the sequence number is assigned when it it enqueued.   The sequence number doesn't represent where in the stream the event is.   It's a constant bit of data that helps to identify that event.   If the third event is now the first one available in the partition, the sequence number doesn't change.